src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php

```diff
@@ -1220,12 +1220,19 @@ private function getProductIdsFromRequest(Request $request): array
     }
 
     /**
-     * @return array
+     * @param string $securitySubject
+     *
+     * @return array<string, array<string, mixed>>
      */
-    private function getProductToolbarButtons(): array
+    private function getProductToolbarButtons(string $securitySubject): array
     {
         $toolbarButtons = [];
 
+        // do not show create button if user has no permissions for it
+        if (!$this->isGranted('create', $securitySubject)) {
```

Comment:
With proper import

```suggestion
        if (!$this->isGranted(Permission::CREATE, $securitySubject)) {
```

Reply:
there is no such class, but good idea to use const. There is PageVoter::CREATE